### DISCUSSION TITLE
I downloaded the code and then it failed to run locally

### DISCUSSION THREAD

**Discussion by 144358** [benbusby/whoogle-search#970] [General]
Pleace ,Can you help me，I'm a Python noob

When using pip deployment or running routes.py file, I set 127.0.0.1:5000, I can't see the search box after opening the web page, I can't see the input content, but when I click search, it prompts :
“Internal Server Error
The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.”

and python idle shell 3.10 show:
"WARNING:app:404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again."

**Comment by benbusby** [comment]
Hi! Do you see any other messages in the shell aside from the "WARNING:app:404" one? That warning message wouldn't be causing the Internal Server Error issue that you're seeing.

Also could you provide details about how you're running whoogle (i.e. what command you're using in the shell)? If you cloned the repo locally, you should be able to run "python3 -um app", but only after you've installed all dependencies via pip.

Edit: If you installed via pip, are you just running `whoogle-search`? Or are you running a different command?

**Comment by 144358** [comment]
> Hi! Do you see any other messages in the shell aside from the "WARNING:app:404" one? That warning message wouldn't be causing the Internal Server Error issue that you're seeing.
> 
> Also could you provide details about how you're running whoogle (i.e. what command you're using in the shell)? If you cloned the repo locally, you should be able to run "python3 -um app", but only after you've installed all dependencies via pip.
> 
> Edit: If you installed via pip, are you just running `whoogle-search`? Or are you running a different command?

Thank you very much. Yes, I only run one program or software at a time. I only use idle to run py files, or use cmd to run after using pip to install whoogle. I haven't found much relevant content about whoogle-search on the network. Few, pip to install whoogle-search, and then run whoogle-search -- host127.0.1 -- port5000; If I run the app folder, routes.py directly, I will prompt an error. I search, modify some function names, and then rerun them. Then they will be displayed in the page after I click the page search box:

“Internal Server Error
The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.”，

 That is, there are no search results. Of course, if there is a detailed successful example for me to watch, such as a video or a picture, I think I will be very happy, and then I can solve the problem. I may need to read more tutorials, or find solutions to those problems, and then I estimate that I will have time to reply and research tomorrow

**Comment by 144358** [comment]
> Hi! Do you see any other messages in the shell aside from the "WARNING:app:404" one? That warning message wouldn't be causing the Internal Server Error issue that you're seeing.
> 
> Also could you provide details about how you're running whoogle (i.e. what command you're using in the shell)? If you cloned the repo locally, you should be able to run "python3 -um app", but only after you've installed all dependencies via pip.
> 
> Edit: If you installed via pip, are you just running `whoogle-search`? Or are you running a different command?

My God, you are the author himself. I am honored to chat with you，hhh

**Comment by benbusby** [comment]
> and then run whoogle-search -- host127.0.1 -- port5000

In this scenario, the host and port aren't required since they're the default values. So you should just be able to run `whoogle-search` by itself and get the same result. In the example you pasted, the spacing of the various elements could cause problems (i.e. "-- host" should be "--host", "-- port" should be "--port"), and each value should have a space after the flag name (i.e. "--port 5000"). This might've only been an issue when writing your comment though.

As for running the repo locally, can you confirm that you performed the following steps:

```bash
git clone https://github.com/benbusby/whoogle-search.git
cd whoogle-search
python3 -m venv venv
source venv/bin/activate
pip install -r requirements.txt
./run
```

If so, can you try running the following command from the whoogle-search directory:

```
python3 -um app --debug
```

And then navigate to localhost:5000 in your browser. If you perform a search and see another error message, can you send the output from your terminal?

These steps should help me narrow down what the problem could be. I'm unfortunately unable to reproduce the issue on my side, so I'm doing the best I can without having access to your actual machine. 

> My God, you are the author himself. I am honored to chat with you，hhh

I'm happy to be chatting with you! Thanks for trying out my project!

**Comment by 144358** [comment]
> > 然后运行 whoogle-search -- host127.0.1 -- port5000
> 
> 在此方案中，主机和端口不是必需的，因为它们是默认值。因此，您应该能够自行运行并获得相同的结果。在您粘贴的示例中，各种元素的间距可能会导致问题（即“-- host”应该是“--host”，“-- port”应该是“--port”），并且每个值都应该在标志名称后有一个空格（即“--port 5000”）。不过，这可能只是在撰写评论时的问题。`whoogle-search`
> 
> 至于在本地运行存储库，您能否确认您执行了以下步骤：
> 
> ```shell
> git clone https://github.com/benbusby/whoogle-search.git
> cd whoogle-search
> python3 -m venv venv
> source venv/bin/activate
> pip install -r requirements.txt
> ./run
> ```
> 
> 如果是这样，您可以尝试从 whoogle-search 目录运行以下命令吗？
> 
> ```
> python3 -um app --debug
> ```
> 
> 然后在浏览器中导航到 localhost：5000。如果您执行搜索并看到另一条错误消息，您可以从终端发送输出吗？
> 
> 这些步骤应该可以帮助我缩小问题的范围。不幸的是，我无法重现该问题，因此我在无法访问您的实际机器的情况下尽我所能。
> 
Ok, I'll try it out based on your comments ，and contact you if there's an update

**Comment by 144358** [comment]
> 

hhh，I'm also happy to talk to you, in my eyes, engineers are great and amazing jobs, making the rules and making everything run according to the rules;

now i go to try the new idea

**Comment by 144358** [comment]
> 
Sorry to bother you again

"""As for running the repository locally, can you confirm that you performed the following steps:
source venv/bin/activate"""

This line of code fails to run in cmd, prompt:
 'source' is not an internal or external command, nor is it a runnable program or batch files. 
Then the other lines are successful, and I need to implement a search page, do I need to run them all? Then I'm now working on how to solve the 'source' problem

I downloaded, "Anaconda3-2022.10-Windows-x86_64", and then conda got git, I think it may be that my computer is missing a lot of modules and causing the failure, tomorrow I will spend part of the time researching this problem, encounter problems that cannot be solved immediately, so that I can learn more new things

I want to know how to build a web page with search function (like: whoogle-search), through Python, implement the specific process, now my idea is :
1, use pip to install, and then run whoogle-search to add nothing;
2, is to run py file I run is routes.py; I think there was an error in my process

**Comment by benbusby** [comment]
I apologize for the delayed reply, I got busy with contracting work and needed to focus on that for a bit. 

Did you run the other steps aside from the `source venv/bin/activate` step (i.e. `pip install -r requirements`)? This isn't completely necessary, but helps to isolate packages required by Whoogle from the rest of your machine. The `pip install -r requirements.txt` step should install all modules that Whoogle needs in order to run successfully.

**Comment by 144358** [comment]
> I apologize for the delayed reply, I got busy with contracting work and needed to focus on that for a bit.
> 
> Did you run the other steps aside from the `source venv/bin/activate` step (i.e. `pip install -r requirements`)? This isn't completely necessary, but helps to isolate packages required by Whoogle from the rest of your machine. The `pip install -r requirements.txt` step should install all modules that Whoogle needs in order to run successfully.

Thank you for replying to me when you are busy, I followed the steps to run until "source venv/bin/activate"," source" does not exist, but I have pip downloaded, "pip install -r requirements.txt" can run, I am not in a hurry about this, now research this matter, it is more like a hobby of mine

**Comment by benbusby** [comment]
> I have pip downloaded, "pip install -r requirements.txt" can run

Can you walk me through step by step and include the console output for each command? Once you've installed all dependencies, you should be able to run `python3 -um app` and then Whoogle will be available on localhost:5000. You might not see anything in the console, but it will be running in the background.

**Reply by 144358** [reply]
Hello, I was thinking when I was not fully asleep in bed, I saw your reply, I am still very happy, my system is using Windows 10 Pro, Python 3.10, now I start to explain the steps I run; 1cmd input: (pip install whoogle-search); 2: input: (git clone https://github.com/benbusby/whoogle-search.git); 3 input: (cd whoogle-search); 4 input: (python3 -m venv venv); None of this prompts an error in cmd; 5 input: (source venv/bin/activate), a prompt appears, I don't think this step was successful, """prompt 'source' is not an internal or external command, nor is it a runnable program
or batch files. “"”； 6 input: (pip install -r requirements.txt), success; 7 input: (./run), indicating that """" is not an internal or external command, nor is it a runnable program
or batch files. """； These are the process of installing with pip, web page I search 127.0.0.1:5000, you can see the home page, the search function cannot be used

**Reply by 144358** [reply]
There are also steps to download zip to run. I will say the following steps: 1 "download zip", "2 decompress", 3 "open the decompression folder", 4 "enter whoogle-search main app, and run routes.py. I use Python's idle and run model, An error prompt will appear (AssertionError: View function mapping is overwriting an existing endpoint function: healthz) ", I rename all the functions defined by def that will report errors", 5. Run the py file again, and there is no error prompt. Enter the webpage http://127.0.0.1:5000/ , the home page appears, and the search box cannot be seen, but the mouse pointer changes, and the search button can be seen. Click the search button to be searched, and prompt：（Internal Server Error The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.）；The results of the two installations are the same,

**Reply by 144358** [reply]
hello，I study temporarily，"https://github.com/144358/AboutWhoogle"，About step picture, cmd input
“whoogle-search"
The results are also relevant; I need to sleep. It's already early morning in my country. I can't think normally soon. Good night

**Comment by benbusby** [comment]
Also, in the meantime, I'm going to convert this issue into a discussion since I think this is a configuration problem rather than a bug with Whoogle.

**Reply by 144358** [reply]
I think too
I'm a little Sorry. I'm afraid it will affect your time too much
